### ISSUE TITLE
Format items as netCDF

### ISSUE THREAD

**Issue by aulemahal** [ECCC-MSC/msc-pygeoapi#88]
This need comes from users of the climatedata.ca website which provides a web interface to make requests to this API. Specifically, I was asked to look into the problem of getting items of the "climate-daily" collection directly in a netCDF format.

AFAIU, a netCDF formatter plugin could be implemented here and activate for items requests? It seems it would be quite straigtforward, adding a xarray (+numpy, pandas) dependency. Would it make sense to this here?

I have time to contribute on this, it would be my first time dealing with pygeoapi.

**Comment by tomkralidis** [comment]
Thanks @aulemahal.    xarray and numpy are already part of coverage processing in core pygeoapi, so looks tractable at first glance.  Looking forward to how the NetCDF/CF output is put together.  Are you thinking of CF point or time series feature types, or something else?

Another option is that if the plugin is generic enough that it can take a result set of GeoJSON and render as NetCDF, it may be a great candidate for pygeoapi core.  For now, let's start here :)

**Comment by aulemahal** [comment]
I never played with this, but it seems that following the "geometries" section of the conventions would make most sense. http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#geometries

For point geometries the result is almost the same as a "normal" timeseries, so it's a good start.

**Comment by aulemahal** [comment]
Hi @tomkralidis. So while starting some work on this, I realized that there was not elegant way fully answer the request I first got through a general formatter. The request is simply too specific. I still think it would be interesting for pygeoapi in general to be able to output netCDFs, but I will not have time to work on this in the near future.

However, as my goal was to return a time series netCDF from the retrieved points, I did write that and if it is of any interest, here's the code: 
https://gist.github.com/aulemahal/24608a6bb8ca901448707f06e7e6dd09

A netCDF formatter for pygeoapi could be implemented starting from this, I think. I'll leave the work for later or someone else.
